### DISCUSSION TITLE
Can somebody help me?!

### DISCUSSION THREAD

**Discussion by sn-o-w** [violentmonkey/violentmonkey#1428] [Q&A]
Hi there,

I use ViolentMonkey for running some userscripts and I have an issue I don't know how to deal with.

I currently use this for one own userscript: [https://github.com/naugtur/insertionQuery](https://github.com/naugtur/insertionQuery)

This works great, I can monitor for DOM changes and it keeps working on detecting the same elements in new pages, but it has a huge disadvantage: I think it would stop working in throttled tabs, sort of.

I'm not sure if what I'm experiencing is called throttle. Maybe it's debouncing? I'm not even sure what these two terms actually mean. Let me explain what I'm actually doing or what I'm trying to do: I use insertionQuery on Twitch, the game live streaming platform, to detect elements appearing on the page.

I can run insertionQuery once in a tab and even if I don't touch that tab for hours, insertionQuery will continue to work, despite the tab being inactive. But only if that page won't switch to a different Twitch page. So animations keep working as long as the page won't change, despite the tab being inactive/idle.

But... and here's the tricky part... Sometimes, Twitch streamers can raid other streamers. This means that without my intervention/input, the tab will switch to another page and insertionQuery will stop working until I go manually back to the respective tab. When I go back to the tab, the script instantly "resumes", and I actually see animations in a flash. I think the new page is actually loaded even if it's not in focus, but somehow animations are stopped until I go back to the inactive/idle tab?!

What really happens and why? It's a limitation of Chrome? And if there's a way to avoid that?! I cannot check so often the tabs insertionQuery runs in, that's why I want to make it somehow keep working in inactive/idle tabs, if possible.

So in a nutshell:
- insertionQuery will run indefinitely even if the tab is idle/inactive for more than 10 minutes and the tab won't switch to another URL;
- insertionQuery will stop running if the tab is idle/inactive for more than 10 minutes AND the tab switched without user intervention/input to another Twitch page.

I also tried to add "@grant window.focus" and "@grant window.onurlchange" in userscript page within Violentmonkey but no use.

Thank you in advance.
